python/mosaic/api/fuse.py

```diff
@@ -0,0 +1,249 @@
+from dataclasses import dataclass
+
+import os
+import pkg_resources
+import requests
+import shutil
+
+__all__ = ["SetupMgr", "setup_fuse_install"]
+
+def get_install_mosaic_version() -> str:
+    """
+    Currently installed version of mosaic.
+
+    Returns
+    -------
+    Installed version of package 'databricks-mosaic' if available;
+    otherwise, None
+    """
+    try:
+        return pkg_resources.get_distribution("databricks-mosaic").version
+    except Exception:
+        pass
+    return None
+
+@dataclass
+class SetupMgr:
+    """
+    Defaults mirror setup_gdal.
+    """
+    to_fuse_dir: str
+    script_in_name: str = 'mosaic-gdal-init.sh'
+    script_out_name: str = 'mosaic-gdal-init.sh'
+    with_mosaic_pip: bool = False
+    with_gdal: bool = True
+    with_ubuntugis: bool = False 
+    override_mosaic_version: str = None
+    jar_copy: bool = False
+    jni_so_copy: bool = False
+
+    def configure(self) -> None:
+        """
+        Handle various config options.
+        - if `with_mosaic_pip` or `with_gdal` or `with_ubuntugis`,
+          script will be configured and written.
+        """
+        # - set the mosaic and github versions
+        #   will be used in downloading resources
+        #   may be used in pip install 
+        mosaic_version = get_install_mosaic_version()
+        github_version = mosaic_version # <- valid or None
+        if (
+                self.override_mosaic_version is not None and 
+                set(self.override_mosaic_version).issubset(set('=0123456789.'))
+        ):
+            github_version = self.override_mosaic_version.replace('=','') 
+        elif mosaic_version is None:
+            github_version = 'main'
+
+        # TODOS AFTER PR MERGED: 
+        # [1] CHANGE URL TO ACTUAL MOSAIC (not 'mjohns-databricks'):
+        #     'https://raw.githubusercontent.com/databrickslabs/mosaic'
+        # [2] USE f'{GITHUB_CONTENT_URL_BASE}/{github_version}' (not 'gdal-jammy-1')
+        GITHUB_CONTENT_URL_BASE = 'https://raw.githubusercontent.com/mjohns-databricks/mosaic'
+        GITHUB_CONTENT_TAG_URL = f'{GITHUB_CONTENT_URL_BASE}/gdal-jammy-1'
+
+        # - generate fuse dir path
+        os.makedirs(self.to_fuse_dir, exist_ok=True)
+
+        with_script = self.with_mosaic_pip or self.with_gdal
+        script_out_path = f'{self.to_fuse_dir}/{self.script_out_name}'
+        if with_script:
+            # - start with the unconfigured script
+            script_url = f'{GITHUB_CONTENT_TAG_URL}/scripts/{self.script_in_name}'
+            script = requests.get(script_url, allow_redirects=True).text
+            
+            # - tokens used in script
+            SCRIPT_FUSE_DIR_TOKEN= "FUSE_DIR='__FUSE_DIR__'"                               # <- ' added
```

Comment:
@mjohns-databricks wouldn't this be simpler with f-strings?
ie `SCRIPT_FUSE_DIR_TOKEN= f"FUSE_DIR='{self.to_fuse_dir}'"`
there would not be a need for if branching below.
Or am I amissing something?

Reply:
ignore me, I got it now.